parse_emails/parse_emails.py

```diff
@@ -141,12 +141,20 @@ def parse(self):
             else:
 
                 raise Exception(f"Unknown file format: [{self._file_type}] for file: [{self._file_name}]")
-            output = recursive_convert_to_unicode(output)
-            self.parsed_email = output
-            return output
+            outputs = recursive_convert_to_unicode(output)
+            outputs = [remove_unicode_spaces(output) for output in outputs] if isinstance(outputs, list) else remove_unicode_spaces(outputs)
+            self.parsed_email = outputs
+            return outputs
 
         except Exception as ex:
             raise Exception(str(ex) + "\n\nTrace:\n" + traceback.format_exc())
+    
+
+def remove_unicode_spaces(output):
+    to_replace = ['\\u200a', '\\u200d']
+    for replace in to_replace:
+        output = {key: val.replace(replace, '') if isinstance(val, str) else val for key, val in output.items()}
+    return output
```

Comment:
how deep is this data structure? does this function need to be recursive like recursive_convert_to_unicode?

Reply:
each call should be with 1 email, in case or attachments the function will be called 1 time for each attachment + the main email.